packages/entityanalytics_ad/data_stream/user/elasticsearch/ingest_pipeline/entity.yml

```
@@ -283,6 +336,54 @@ processors:
           field: "_ingest._value.id"
           value: "{{{_ingest._value.object_sid}}}"
 
+  - script:
+      lang: painless
+      ignore_failure: true
+      tag: set_privileged_group_member
+      description: Set privileged_group_member flag based on group membership.
+      # Well-known SIDs - Name to SID mapping
+      # https://learn.microsoft.com/en-us/windows-server/identity/ad-ds/manage/understand-security-identifiers#well-known-sids
+      source: >
+        if (ctx.activedirectory?.groups == null) {
+          return;
+        }
+        ctx.activedirectory.user.privileged_group_member = false;
+        for (def group : ctx.activedirectory?.groups) {
+          if (group?.object_sid != null && (
+            // Domain Admins
+            group.object_sid.endsWith('-512') ||
+            // Domain Controllers
+            group.object_sid.endsWith('-516') ||
+            // Schema Admins
+            group.object_sid.endsWith('-518') ||
+            // Enterprise Admins
+            group.object_sid.endsWith('-519') ||
+            // Group Policy Creator Owners
+            group.object_sid.endsWith('-520') ||
+            // Protected Users
+            group.object_sid.endsWith('-525') ||
+            // Key Admins
+            group.object_sid.endsWith('-526') ||
+            // Enterprise Key Admins
+            group.object_sid.endsWith('-527') ||
+            // Administrators
+            group.object_sid.endsWith('-544') ||
+            // Account Operators
+            group.object_sid.endsWith('-548') ||
+            // Server Operators
+            group.object_sid.endsWith('-549') ||
+            // Backup Operators
+            group.object_sid.endsWith('-551'))) {
+              ctx.activedirectory.user.privileged_group_member = true;
+              break; // Stop at a match
+          }
```

```suggestion
          if (group?.object_sid == null) {
            continue;
          }
          int idx = group.object_sid.lastIndexOf('-');
          if (idx < 0) {
            continue;
          }
          def priv = params.get(group.object_sid.substring(idx+1);
          if (priv != null) {
            ctx.activedirectory.user.privileged_group_member = priv;
            return;
          }
```
with the suffixes in a params table without the dash.

```
params:
  '512': true
  '516': true
  '518': true
  '519': true
  '520': true
  '525': true
  '526': true
  '527': true
  '544': true
  '548': true
  '549': true
  '551': true
```

---

Done